src/features/marketplace/components/rewards/ItemDetail.tsx

```diff
@@ -0,0 +1,191 @@
+  const [showSuccess, setShowSuccess] = useState<boolean>(false);
+  const [confirmBuy, setConfirmBuy] = useState(false);
+  const [
+    {
+      context: { state },
+    },
+  ] = useActor(gameService);
+  const { ingredients, items, image, description } =
+    TRADE_REWARDS(state)[itemName];
+
+  useLayoutEffect(() => {
+    const imgElement = new Image();
+
+    imgElement.onload = function () {
+      const trueWidth = imgElement.width;
+      const scaledWidth = trueWidth * PIXEL_SCALE;
+
+      setImageWidth(scaledWidth);
+    };
+
+    imgElement.src = image;
+  }, [image]);
+
+  const { inventory } = state;
+
+  const canBuy = () =>
+    inventory["Trade Point"]?.gte(ingredients["Trade Point"]) &&
+    !hasHitInventoryLimit;
+
+  const getButtonLabel = () => {
+    if (confirmBuy) return `${t("confirm")} ${t("buy")}`;
+
+    return `${t("buy")}`;
+  };
+
+  const buttonHandler = () => {
+    if (!confirmBuy) {
+      setConfirmBuy(true);
+      return;
+    }
+
+    handleBuy();
+  };
+
+  const handleBuy = () => {
+    gameService.send("reward.redeemed", {
+      item: itemName,
+    });
+
+    if (showAnimations) confetti();
+    setShowSuccess(true);
+    setConfirmBuy(false);
+  };
+
+  const hasHitInventoryLimit =
+    itemName === "Seed Pack" &&
+    getKeys(items).some((item) => {
+      const inventoryAmount = state.inventory[item] ?? new Decimal(0);
+      const inventoryLimit = INVENTORY_LIMIT(state)[item];
+
+      if (!!inventoryLimit && inventoryAmount.gte(inventoryLimit)) {
+        return true;
+      }
+    });
+
+  return (
+    <CloseButtonPanel onClose={onClose}>
+      {!showSuccess && (
+        <>
+          <div className="w-full p-2 px-1">
+            <div className="flex">
+              <div
+                className="w-[50%] relative min-w-[40%] rounded-md overflow-hidden shadow-md mr-2 flex justify-center items-center h-32"
+                style={{
+                  backgroundImage: `url(${SUNNYSIDE.ui.grey_background})`,
+                  backgroundSize: "cover",
+                  backgroundPosition: "center",
+                }}
+              >
+                <img
+                  src={image}
+                  alt={itemName}
+                  className={"w-full"}
+                  style={{
+                    width: `${imageWidth}px`,
+                  }}
+                />
+              </div>
+              <div className="flex flex-col space-y-2">
+                <Label className="text-sm" type={"default"}>
+                  {itemName}
+                </Label>
+                <p>{description}</p>
+                <p>{"Contains:"}</p>
+                <div className="flex flex-row flex-wrap">
+                  {getKeys(items).map((item, index) => {
+                    if (items[item] === 0) return;
+                    return (
+                      <Box
+                        key={index}
+                        image={ITEM_DETAILS[item].image}
```

Comment:
I'm getting test errors as a result of this line here. I'm not too sure how to resolve it